src/Apps/Order2/Routes/Checkout/Components/PaymentStep/Order2PaymentForm.tsx

```diff
@@ -409,9 +476,25 @@ const PaymentFormContent: React.FC<PaymentFormContentProps> = ({
         return
       }
 
+      const billingAddress = getBillingAddress()
       const { error, confirmationToken } = await stripe.createConfirmationToken(
         {
           elements,
+          params: {
```

Comment:
This will be executed for all stripe payment methods, so we need to refactor this a bit to make billing optional and only pass it in the condition below
`if (selectedPaymentMethod === "stripe-card")`